src/azure-cli/azure/cli/command_modules/acr/repository.py

```diff
@@ -284,6 +285,12 @@ def acr_repository_update(cmd,
         json_payload.update({
             'writeEnabled': write_enabled
         })
+    if teleport_enabled is not None:
+        if not repository:
+            raise CLIError("--teleport-enabled can only be used with --repository.")
```

Comment:
recommend to use error type in azclierror, such as 
**https://github.com/Azure/azure-cli/blob/3eea7fd773722b7a3c40b8115f29a3bad39b70bc/src/azure-cli-core/azure/cli/core/azclierror.py#L157-L161**